libexec/rtld-elf/rtld_c18n.h

```diff
@@ -208,10 +210,20 @@ func_sig_legal(struct func_sig sig)
 /*
  * APIs
  */
+#define	c18n_return_address()	(C18N_ENABLED ?				\
+	_Pragma("clang diagnostic push")				\
+	_Pragma("clang diagnostic ignored \"-Wframe-address\"")		\
+	((struct trusted_frame *)__builtin_frame_address(1))->pc :	\
+	_Pragma("clang diagnostic pop")					\
+	__builtin_return_address(0))
```

Comment:
This might be more readable as a statement expression. https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html